tests/integration/test_gcloud/test_functioning/test_day_wday.py

```diff
@@ -0,0 +1,191 @@
+
+p, m = generate_params(extra_params=conf_params,
+                       modes=monitoring_modes)
+
+configurations = load_wazuh_configurations(configurations_path, __name__, params=p, metadata=m)
+
+
+# fixtures
+
+@pytest.fixture(scope='module', params=configurations)
+def get_configuration(request):
+    """Get configurations from the module."""
+    return request.param
+
+
+# tests
+
+@pytest.mark.parametrize('tags_to_apply', [
+    ({'ossec_day_conf'}),
+    ({'ossec_wday_conf'}),
+    ({'ossec_time_conf'})
+])
+@pytest.mark.skipif(sys.platform == "win32", reason="Windows does not have support for Google Cloud integration.")
+def test_day_wday(tags_to_apply, get_configuration, configure_environment,
+                  restart_wazuh, wait_for_gcp_start):
+    """
+    These tests verify the module starts to pull according to the day of the week 
+    or month and time.
+    """
+    check_apply_test(tags_to_apply, get_configuration['tags'])
+
+    with pytest.raises(TimeoutError):
+        event = wazuh_log_monitor.start(timeout=3,
+                                        callback=callback_detect_start_fetching_logs).result()
+        raise AttributeError(f'Unexpected event {event}')
+
+    wazuh_log_monitor.start(timeout=global_parameters.default_timeout + 120,
+                            callback=callback_detect_start_fetching_logs,
+                            accum_results=1,
+                            error_message='Did not receive expected '
+                                          '"Starting fetching of logs" event').result()
+    next_scan_time_log = wazuh_log_monitor.start(timeout=global_parameters.default_timeout + 60,
+                                                 callback=callback_detect_start_gcp_sleep,
+                                                 accum_results=1,
+                                                 error_message='Did not receive expected '
+                                                               '"Sleeping until ..." event').result()
+    next_scan_time_spl = next_scan_time_log.split(" ")
+    date = next_scan_time_spl[0].split("/")
+    hour = next_scan_time_spl[1].split(":")
+
+    test_now = datetime.datetime.now()
+    next_scan_time = datetime.datetime(int(date[0]), int(date[1]), int(date[2]), int(hour[0]), int(hour[1]),
+                                       int(hour[2]))
+    diff_time = (next_scan_time - test_now).total_seconds()
+    seconds = (int(diff_time - 20))
+
+    TimeMachine.travel_to_future(timedelta(seconds=seconds))
+
+    test_today = datetime.date.today()
+    if tags_to_apply == 'ossec_day_conf':
+        if today.month < 12:
+            assert test_today.month == today.month + 1
+        else:
+            assert test_today.month == 1
+    if tags_to_apply == 'ossec_wday_conf':
+        assert weekDays[test_today.weekday()] == wday
+        assert test_today.day == day + datetime.timedelta(weeks=1)
```

Comment:
We expect `day` to be `int`, and `datetime.timedelta(weeks=1)` to be `<class 'datetime.date'>`.
Addition of `int` and `<class 'datetime.date'>` seems not to be supported.

Reply:
Fixed